### ISSUE TITLE
[oracle] ALTER TABLE ADD CONSTRAINT with bracketed arguments is not parsable

### ISSUE THREAD

**Issue by Ralf0711** [sqlfluff/sqlfluff#8075]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

We have a lot of files with this content (Example):

```
ALTER TABLE ACS_DPS_STRUKTURKLASSE ADD (
  CONSTRAINT DPSS_DPSI_EID_FK FOREIGN KEY (EXTACSGROUP_ID) 
    REFERENCES ACS_DPSICHT (EXTACSGROUP_ID));
```

### Expected Behaviour

The execution of `sqlfluff lint --dialect oracle` should not WARN/FAIL for these file contents.

The Workaround is to delete the enclosing brackets ().
```
ALTER TABLE ACS_DPS_STRUKTURKLASSE ADD 
  CONSTRAINT DPSS_DPSI_EID_FK FOREIGN KEY (EXTACSGROUP_ID) 
    REFERENCES ACS_DPSICHT (EXTACSGROUP_ID);
```

### Observed Behaviour

We execute `sqlfluff lint --dialect oracle`
-> sqlfluff lint result is:
```
L:   4 | P:   1 |  PRS | WARNING: Line 4, Position 1: Found unparsable section:
                       | 'ALTER TABLE ACS_DPS_STRUKTURKLASSE ADD (...'
```

### How to reproduce

1. Create a sql-script with this content
```
ALTER TABLE ACS_DPS_STRUKTURKLASSE ADD (
  CONSTRAINT DPSS_DPSI_EID_FK FOREIGN KEY (EXTACSGROUP_ID) 
    REFERENCES ACS_DPSICHT (EXTACSGROUP_ID));
```
2. execute `sqlfluff lint --dialect oracle`

### Dialect

oracle

### Version

```
$ sqlfluff --version   
sqlfluff, version 4.2.2
```

### Configuration

no customized config

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
